src/app/firedancer-dev/commands/snapshot_load.c

```diff
@@ -216,6 +220,149 @@ fsck_vinyl( config_t * config ) {
+    cnt += hist->bin_cnt[ i ];
+    acc += hist->bin_acc[ i ];
+  }
+  if( cnt != hist->total_cnt ) return -1;
+  if( acc != hist->total_acc ) return -2;
+  return 0;
+}
+
+static
+void accounts_hist_print( accounts_hist_t const * hist,
+                          char const            * description ) {
+  double hist_total_cnt_M   = (double)hist->total_cnt / (double)1.0e6;
+  double hist_total_cnt_GiB = (double)hist->total_acc / (double)1073741824;
+  printf( "\n" );
+  printf( "Accounts sizes histogram %s\n", description );
+  printf( "hist_total_cnt %16lu ( %6.1f M   )\n", hist->total_cnt, hist_total_cnt_M   );
+  printf( "hist_total_acc %16lu ( %6.1f GiB )\n", hist->total_acc, hist_total_cnt_GiB );
+  printf( "   bin_th_lo ?? sz ??    bin_th_hi |    bin_cnt (run_sum%%) |      bin_acc (run_sum%%) |    bin_min B |    bin_max B |    bin_avg B |\n" );
+  ulong sum_cnt = 0UL;
+  ulong sum_acc = 0UL;
+  for( int i=0; i < ACCOUNTS_HIST_N; i++ ) {
+    /* bin thresholds */
+    ulong hist_bin_tlo      = hist->bin_thi[ fd_int_if( i>0, i-1, i ) ];
+    ulong hist_bin_thi      = hist->bin_thi[ i ];
+    /* bin cnt */
+    ulong hist_bin_cnt      = hist->bin_cnt[ i ];
+    sum_cnt                += hist->bin_cnt[ i ];
+    double sum_cnt_p        = (double)(sum_cnt * 100) / (double)hist->total_cnt;
+    double hist_bin_cnt_K   = (double)(hist_bin_cnt) / (double)1.0e3;
+    /* bin acc */
+    ulong hist_bin_acc      = hist->bin_acc[ i ];
+    sum_acc                += hist->bin_acc[ i ];
+    double sum_acc_p        = (double)(sum_acc * 100) / (double)hist->total_acc;
+    double hist_bin_acc_MiB = (double)(hist_bin_acc) / (double)1048576.0f;
+    /* bin min, max, avg */
+    ulong hist_bin_min      = fd_ulong_if( hist->bin_cnt[ i ] > 0, hist->bin_min[ i ], 0UL );
+    ulong hist_bin_max      = hist->bin_max[ i ];
+    ulong hist_bin_avg      = fd_ulong_if( hist->bin_cnt[ i ] > 0, hist->bin_acc[ i ] / hist->bin_cnt[ i ], 0UL );
+    /* log */
+    char buf[256];
+    char * p = fd_cstr_init( buf );
+    p = fd_cstr_append_printf( p, "%12lu %s sz <= %12lu |", hist_bin_tlo, i==0? "<=" : "< ", hist_bin_thi );
+    p = fd_cstr_append_printf( p, " %8.1f K (%6.1f %%) |", hist_bin_cnt_K, sum_cnt_p );
+    p = fd_cstr_append_printf( p, " %8.1f MiB (%6.1f %%) |", hist_bin_acc_MiB, sum_acc_p );
+    p = fd_cstr_append_printf( p, " %12lu | %12lu | %12lu |", hist_bin_min, hist_bin_max, hist_bin_avg );
+    p = fd_cstr_append_printf( p, "\n" );
+    printf( "%s", buf );
+  }
+  printf( "\n" );
+}
+
+static inline void
+accounts_hist_vinyl( accounts_hist_t * hist,
+                     fd_snapin_tile_t * ctx ) {
+  for( ulong ele_i=0; ele_i < fd_vinyl_meta_ele_max( ctx->vinyl.map ); ele_i++ ) {
+    fd_vinyl_meta_ele_t const * ele = ctx->vinyl.map->ele + ele_i;
+    if( FD_UNLIKELY( fd_vinyl_meta_private_ele_is_free( ctx->vinyl.map->ctx, ele ) ) ) continue;
+    accounts_hist_update( hist, (ulong)ele->phdr.info._val_sz );
+  }
+}
+
+static inline void
```

Comment:
```suggestion
static void
```

These are pretty big functions I wouldn't make them inline